boot/bootutil/src/loader.c

```diff
@@ -781,9 +782,16 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
 #if defined(MCUBOOT_OVERWRITE_ONLY) && defined(MCUBOOT_DOWNGRADE_PREVENTION)
     if (slot != BOOT_PRIMARY_SLOT) {
         /* Check if version of secondary slot is sufficient */
-        rc = boot_version_cmp(
-                &boot_img_hdr(state, BOOT_SECONDARY_SLOT)->ih_ver,
-                &boot_img_hdr(state, BOOT_PRIMARY_SLOT)->ih_ver);
+
+	/* Prototype to be moved in to hook */
+	#if defined(CONFIG_SOC_NRF5340_CPUAPP)
+	 rc = boot_version_cmp_net(BOOT_CURR_IMG(state), boot_img_hdr(state, BOOT_SECONDARY_SLOT)->ih_ver,
+			           &boot_img_hdr(state, BOOT_PRIMARY_SLOT)->ih_ver);
+	#else
+	rc = boot_version_cmp(
+			&boot_img_hdr(state, BOOT_SECONDARY_SLOT)->ih_ver,
+			&boot_img_hdr(state, BOOT_PRIMARY_SLOT)->ih_ver);
+	#endif
```

Comment:
This doesn't look right, this will mess up someone marking the secondary slot of the primary image for test/confirmation because for some reason it will check the network core version, despite not having anything to do with the network core? And only applies if MCUboot is in upgrade mode?